nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/tools/command/PositionCommand.java

```diff
@@ -0,0 +1,146 @@
+        options.addOption(Option.builder()
+                .longOpt("to-end")
+                .desc("Sets the current position to the end for all partitions")
+                .build());
+        options.addOption(Option.builder()
+                .longOpt("to-timestamp")
+                .desc("Sets the current position to a record >= timestamp eg. " + Instant.now())
+                .hasArg()
+                .argName("TIMESTAMP")
+                .build());
+    }
+
+    @Override
+    public boolean run(LogManager manager, CommandLine cmd) throws InterruptedException {
+        String name = cmd.getOptionValue("log-name");
+        String group = cmd.getOptionValue("group", "tools");
+
+        if (cmd.hasOption("to-timestamp")) {
+            long timestamp = parseTimestamp(cmd.getOptionValue("to-timestamp"));
+            if (timestamp < 0) {
+                return false;
+            }
+            positionToTimestamp(manager, group, name, timestamp);
+        } else if (cmd.hasOption("to-end")) {
+            toEnd(manager,group,name);
+        } else if (cmd.hasOption("reset")) {
+            reset(manager,group,name);
+        } else {
+            System.err.println("Invalid option, try 'help position'");
+            return false;
+        }
+
+        return true;
+    }
+
+    protected void toEnd(LogManager manager, String group, String name) {
+        LogLag lag = manager.getLag(name, group);
+
+        try (LogTailer<Record> tailer = manager.createTailer(group, name)) {
+            tailer.toEnd();
+        }
+        System.out.println(String.format("# Moved log %s, group: %s, from: %s to %s", name, group, lag.lag(), lag.upper()));
+    }
+
+    protected void reset(LogManager manager, String group, String name) {
+        LogLag lag = manager.getLag(name, group);
+        long pos = lag.lower();
+        try (LogTailer<Record> tailer = manager.createTailer(group, name)) {
+            tailer.reset();
+        }
+        System.out.println(String.format("# Reset log %s, group: %s, from: %s to 0", name, group, pos));
+    }
+
+    protected void positionToTimestamp(LogManager manager, String group, String name, long timestamp) {
+
+        try (LogTailer<Record> tailer = manager.createTailer(group, name)) {
+            int size = manager.getAppender(name).size();
+            IntStream.range(0, size).boxed().map(partition -> new LogPartition(name, partition)).forEach(lp -> {
+                LogOffset logOffset = tailer.offsetForTimestamp(lp, timestamp);
+                if (logOffset != null) {
+                    tailer.seek(logOffset);
+                    tailer.commit();
+                    System.out.println(
+                            String.format("# Set log %s, group: %s, to offset %s", name, group, logOffset.offset()));
+                } else {
+                    System.err.println(
+                            String.format("# Could not find an offset for log %s, group: %s", name, group));
```

Comment:
if the command fails (timestamp is not found or invalid) run method should return false (which will generate an error exit code), you can also use assertion on the returned state in unit tests.